docs/development.md

```diff
@@ -281,12 +282,55 @@ I0511 10:15:29.593106 244858 hello_world.cc:57] Response: HELLO AGAIN WORLDS!
 
 ### Signing Oak Wasm modules
 
-Oak Wasm modules can be signed using [Ed25519](https://ed25519.cr.yp.to/)
-scheme.
+As described in [Oak Concepts](/docs/concepts.md#labels) Oak Wasm modules can be
+signed using [Ed25519](https://ed25519.cr.yp.to/) scheme.
 
-In order create a signature `hello_world.sign` for a Wasm module
-`hello_world.wasm` use the following script:
+If you are reviewing an Oak module (for example `hello_world.wasm`) and want to
+create a signature for it, first you need a private/public key pair. It can be
+generated using `oak_sign` utility:
 
 ```bash
-./scripts/sign_module -i examples/hello_world/bin/hello_world.wasm -o examples/hello_world/bin/hello_world.sign
+./scripts/oak_sign generate \
+  --private-key=examples/certs/ed25519/test.key \
+  --public-key=examples/certs/ed25519/test.pub
 ```
+
+It generates two key files (`test.key` and `test.pub`) that are encoded using
+[PEM](https://tools.ietf.org/html/rfc1421) format.
+
+Then a Wasm module can be signed using the following command:
+
+```bash
+./scripts/oak_sign sign \
+  --private-key=examples/certs/ed25519/test.key \
+  --input-file=examples/hello_world/bin/hello_world.wasm \
+  --signature=examples/hello_world/bin/hello_world.sign
+```
+
+This command generates a signature file `hello_world.sign` contains both a
+public key and a Wasm modules signature (also encoded with PEM).
+
+Wasm module signatures should be passed to `oak_loader` by an Oak server
+maintainer using the `--signatures-manifest=signature.toml`, where
+`signature.toml` file contains paths to signature files for each signed Oak Wasm
+module (each module is referred by its SHA-256 hash). Also, since each module
+can be signed by multiple entities, each module may have multiple signatures.
+
+The `signature.toml` file can look like this:
+
+```toml
+[signatures]
+d584ac26782b91e5d47f35a1e0eca6836b7b6aaa09e624a3baea09b6f01ffe5b = [
+  { path = "examples/hello_world/bin/hello_world.sign" },
+]
```

Comment:
I commented on the other PR, but I really think that the hash should be part of the signature. If a reviewer signs a Wasm module, the resulting file should have all the information necessary to link the signature, public key and module identifier. In the current form, imagine I sign something, I generate `hello_world.sign`, now I send it to you, and the file itself is not sufficient, I also need to tell you (in some other way) what the hash is. May as well make it part of the file, no? Again, if this were a mySQL table, it would have 3 columns I imagine, so it makes sense that these three pieces of information are part of a single entry.

Reply:
Updated